Classes/Providers/OTT/PhoenixMediaProvider.swift

```diff
@@ -0,0 +1,395 @@
+//
+//  OTTEntryProvider.swift
+//
+//
+//  Created by Admin on 13/11/2016.
+//
+//
+
+import UIKit
+import SwiftyJSON
+
+/************************************************************/
+// MARK: - PhoenixMediaProviderError
+/************************************************************/
+public enum PhoenixMediaProviderError: PKError {
+
+    case invalidInputParam(param: String)
+    case unableToParseData(data: Any)
+    case noSourcesFound
+    case serverError(info:String)
+
+    static let domain = "com.kaltura.playkit.error.PhoenixMediaProvider"
+
+    var code: Int {
+        switch self {
+        case .invalidInputParam: return 0
+        case .unableToParseData: return 1
+        case .noSourcesFound: return 2
+        case .serverError: return 3
+        }
+    }
+
+    var errorDescription: String {
+
+        switch self {
+        case .invalidInputParam(let param): return "Invalid input param: \(param)"
+        case .unableToParseData(let data): return "Unable to parse object"
+        case .noSourcesFound: return "No source found to play content"
+        case .serverError(let info): return "Server Error: \(info)"
+        }
+    }
+
+    var userInfo: [String: Any] {
+        return [String: Any]()
+    }
+
+}
+
+/************************************************************/
+// MARK: - PhoenixMediaProvider
+/************************************************************/
+
+/* Description
+ 
+    Using Session provider will help you create MediaEntry in order to play content with the player
+    It's requestig the asset data and creating sources with relevant information for ex' contentURL, licenseURL, fiarPlay certificate and etc'
+ 
+    #Example of code
+    ````
+    let phoenixMediaProvider = PhoenixMediaProvider()
+    .set(type: AssetType.media)
+    .set(assetId: asset.assetID)
+    .set(fileIds: [file.fileID.stringValue])
+    .set(networkProtocol: "https")
+    .set(playbackContextType: isTrailer ? PlaybackContextType.trailer : PlaybackContextType.playback)
+    .set(sessionProvider: PhoenixSessionManager.shared)
+
+    phoenixMediaProvider.loadMedia(callback: { (media, error) in
+    
+    if let mediaEntry = media, error == nil {
+        self.player?.prepare(MediaConfig.config(mediaEntry: mediaEntry, startTime: params.startOver ? 0 : asset.currentMediaPositionInSeconds))
+    }else{
+        print("error loading asset: \(error?.localizedDescription)")
+        self.delegate?.corePlayer(self, didFailWith:LS("player_error_unable_to_load_entry"))
+    }
+    ````
+})
+*/
+@objc public class PhoenixMediaProvider: NSObject, MediaEntryProvider {
+
+    var sessionProvider: SessionProvider?
```

Comment:
Shouldn't all this media providers properties be public? if so we also need to add @objc attribute to them (except for the executor)